ihontaryk/homework_8/warrior.py

```diff
@@ -0,0 +1,50 @@
+class Player:
+    def __init__(self, health, attack):
+        self.health = health
+        self.attack = attack
+
+    @property
+    def is_alive(self):
+        return self.health > 0
+
+    def hit(self, player: 'Player'):
+        player.lose_health(self.attack)
+
+    def lose_health(self, damage: int):
+        self.health = max(0, self.health - damage)
+
+
+class Warrior(Player):
+    def __init__(self, health, attack):
+        super().__init__(health, attack)
+
+
+class Knight(Player):
+    def __init__(self, health, attack):
+        super().__init__(health, attack)
+
+
+def fight(player1: Warrior, player2: Knight):
```

Comment:
fight does not expect Warrior and Knight, it expects any class - i.e. Base class, in you case Player